pkgs/top-level/perl-packages.nix

```diff
@@ -7194,17 +7194,29 @@ with self; {
 
   DBDmysql = buildPerlPackage {
     pname = "DBD-mysql";
-    version = "4.050";
+    version = "5.010";
 
     src = fetchurl {
-      url = "mirror://cpan/authors/id/D/DV/DVEEDEN/DBD-mysql-4.050.tar.gz";
-      hash = "sha256-T0hUH/FaCnQF92rcEPgWJ8M5lvv1bJXCbAlERMCSjXg=";
+      url = "mirror://cpan/authors/id/D/DV/DVEEDEN/DBD-mysql-5.010.tar.gz";
+      hash = "sha256-LKL/Odk+idT3RG5fD68DgF6RZ+6bigS6fLJG4stG7uc=";
     };
 
-    buildInputs = [ pkgs.libmysqlclient DevelChecklib TestDeep TestDistManifest TestPod ];
+    nativeBuildInputs = [
+      pkgs.mysql80 # for mysql_config
+    ];
+    buildInputs = [
+      DevelChecklib
+      TestDeep
+      TestDistManifest
+      TestPod
+      pkgs.libmysqlconnectorcpp
+      pkgs.libxcrypt
+      pkgs.openssl
+      pkgs.zstd
+    ];
```

Comment:
So, I forget the details of the MySQL/MariaDB split, but: previously we were using `libmysqlclient`, which is actually `mariadb-connector-c`. Now we’re pulling in an entire build of actual MySQL and `libmysqlconnectorcpp` which also seems to be a MySQL thing. I’m not saying this is necessarily a problem but I want to make sure we want to include MySQL in the default desktop closure, and indeed the closure of `mariadb-server` itself?

There’s also `DBD::MariaDB` – I don’t know if it’s compatible with `mytop` or not, but maybe we should be using that instead? cc @Conni2461 @dasJ

Reply:
I think it is  needed for `mytop.sh` and https://github.com/MariaDB/server/blob/eff9c198e32a828f610b93fad3a0f0eb63b3ded2/scripts/mytop.sh#L250-L255  seems to show that it is fine with `DBDMariaDB`

Reply:
I can switch it over, but not really test it.

Reply:
it builds. I switched it over a a few days ago when building but thought that `mytop.sh` only supported the `mysql` version so didn't move forward -- looking over it again I misread the `mytop.sh` code, but like @FliegendeWurst implies, testing that it works is the hard part.

Reply:
It would be sort of embarrassing if MariaDB were shipping a script that purported to support MariaDB but that actually required MySQL to build, I think. It doesn’t look like the script is doing anything fancy with the connection. I’d be comfortable switching to unbreak it.

Reply:
assuming the `DBDMariaDB` perl package works, I agree. the diff is pretty straightforward.

```diff
diff --git a/pkgs/servers/sql/mariadb/default.nix b/pkgs/servers/sql/mariadb/default.nix
index a178e1fa28e4..921510a07f91 100644
--- a/pkgs/servers/sql/mariadb/default.nix
+++ b/pkgs/servers/sql/mariadb/default.nix
@@ -26,7 +26,7 @@ let
 
     libExt = stdenv.hostPlatform.extensions.sharedLibrary;
 
-    mytopEnv = buildPackages.perl.withPackages (p: with p; [ DBDmysql DBI TermReadKey ]);
+    mytopEnv = buildPackages.perl.withPackages (p: with p; [ DBDMariaDB DBI TermReadKey ]);
 
     common = rec { # attributes common to both builds
       inherit version;
```